corehq/apps/reports/util.py

```diff
@@ -602,3 +608,46 @@ def _add_new_user_to_HQ_group(session, local_user):
         _sync_tableau_users_with_hq(domain)
         # Sync the TableauUser model with Tableau users on the remote Tableau instance
         _sync_tableau_users_with_remote(domain)
+
+
+# Attaches to the parse_web_users method
+def add_on_tableau_details(domain, web_user_dicts):
+    tableau_users_for_domain = TableauUser.objects.filter(server=TableauServer.objects.get(domain=domain))
+    session = TableauAPISession.create_session_for_domain(domain)
+    for web_user_dict in web_user_dicts:
+        user = CouchUser.get_by_username(web_user_dict['username'], strict=True)
+        # Make sure it's not just an invite
+        if user and user.get_domain_membership(domain):
+            try:
+                tableau_user = tableau_users_for_domain.get(username=user.username)
+                web_user_dict['tableau_role'] = tableau_user.role
+                web_user_dict['tableau_groups'] = json.dumps(_group_json_to_tuples(
+                    session.get_groups_for_user_id(tableau_user.tableau_user_id)
+                ))
```

Comment:
Seems like this could be very slow, assuming there aren't going to be a large number of groups you could potentially invert this to get the list of users in all the groups first and then just check which groups each user is in.